gpkit/substitution.py

```diff
@@ -179,8 +178,7 @@ def substitution(nomial, substitutions, val=None):
                 sub = VarKey(name=sub, **descr)
                 exps_[i] += HashVector({sub: x})
                 varlocs_[sub].append(i)
-            elif isinstance(sub, VarKey):
-                sub = VarKey(sub)
+            elif isinstance(sub, (VarKey, Monomial)):
                 if isinstance(var.units, Quantity):
                     try:
                         new_units = var.units/sub.units
```

Comment:
hmm, `new_units` isn't a great variable name, since it's dimensionless.

in the line below this one, `cs_[i]` gets multiplied by `new_units.to('dimensionless')`.

I think it's correct as written here -- this causes a `var` in `cm` to have it's `cs` multiplied by 0.01 when it is substituted by a `sub` in meters:

```python
In [9]: Q(1, "cm")/Q(1, "m")
Out[9]: <Quantity(1.0, 'centimeter / meter')>

In [10]: _.to("dimensionless")
Out[10]: <Quantity(0.01, 'dimensionless')>
```